src/com/google/javascript/jscomp/CompilerOptionsPreprocessor.java

```diff
@@ -43,14 +43,14 @@ static void preprocess(CompilerOptions options) {
 
     if (options.getLanguageIn() == options.getLanguageOut()) {
       // No conversion.
-    } else if (!options.getLanguageIn().isEs6OrHigher()) {
+    } else if (!options.getLanguageIn().isEs6OrHigher() && !options.allowEs6ToEs6) {
```

Comment:
I don't think we need this. This is to prevent things like --language_in=ES5 --language_out=ES3